Resources/public/js/components/products/components/attributes/main.js

```diff
@@ -0,0 +1,365 @@
+                    }
+
+                }.bind(this));
+
+            }.bind(this))
+            .done(function() {
+                // create container for overlay
+                var $overlay = this.sandbox.dom.createElement('<div>');
+                this.sandbox.dom.append(this.$el, $overlay);
+
+                // create content
+                this.sandbox.start([
+                    {
+                        name: 'overlay@husky',
+                        options: {
+                            el: $overlay,
+                            supportKeyInput: false,
+                            title: this.sandbox.translate('product.attribute.overlay.title'),
+                            skin: 'normal',
+                            openOnStart: true,
+                            removeOnClose: true,
+                            instanceName: overlayInstanceName,
+                            data: createAddOverlayContent.call(this),
+                            okCallback: overlayOkClicked.bind(this)
+                        }
+                    }
+                ]);
+
+            }.bind(this))
+            .fail(function() {
+                console.log("Error retrieving attributes from server");
+            });
+
+            jqxhr.complete(function() {
+                // create dropbox in overlay
+                var selectOptions = {
+                    el: '#selectBox',
+                    instanceName: selectInstanceName,
+                    multipleSelect: false,
+                    defaultLabel: this.sandbox.translate('product.attribute.overlay.defaultlabel'),
+                    valueName: 'name',
+                    isNative: true,
+                    data: this.attrTypes
+                };
+
+                this.sandbox.start([
+                    // start select
+                    {
+                        name: 'select@husky',
+                        options: selectOptions
+                    }
+                ]);
+
+                // define select event for dropbox
+                this.sandbox.on('husky.select.' + selectInstanceName + '.selected.item', function(item) {
+                    attrId = parseInt(item);
+                });
+            }.bind(this));
+
+        },
+
+        /**
+         * save product attributes
+         */
+        save = function() {
+            this.sandbox.emit('sulu.products.save', this.sendData);
+        },
+
+        /**
+         * called when OK on overlay was clicked
+         */
+        overlayOkClicked = function() {
+
+            this.sendData = new Object();
+            var attributeValue = this.sandbox.dom.val('#attribute-name');
+
+            var attributes = this.options.data.attributes;
+
+            var result = _.findWhere(attributes, {'attributeId': attrId});
+
+            if (result) {
+                result.value = attributeValue;
+                // update action = 3
+                this.sendData.action = 3;
```

Comment:
you could also use strings instead of numbers, which would be much more readable.

```{javascript}
this.sendData.action = 'UPDATE'
```
or define it as variable at the beginning of the file:

```{javascript}
  var actions = {
    ADD: 1,
    DELETE: 2
    UPDATE: 3
  }

  ...

this.sendData.action =  actions.UPDATE;
```